src/api/routes/gem2s.js

```diff
@@ -34,17 +34,18 @@ module.exports = {
     }
 
     const { io, parsedMessage } = result;
-
-    try {
-      await gem2sResponse(io, parsedMessage);
-    } catch (e) {
-      logger.error(
-        'Pipeline response handler failed with error: ', e,
-      );
-
-      AWSXRay.getSegment().addError(e);
-      res.status(200).send('nok');
-      return;
+    if (parsedMessage) {
```

Comment:
Does this mean that if `parsedMessage` doesn't exist, the function will return 200? Would it be better to throw something which says `parsedMessage` doesn't exist?

Reply:
The absence of parsed message does not mean that an error has occurred: `parseSNSMessage` returns a  `{ io, parsedMessage }` if its input has `Type == Notification`; otherwise (`SubscriptionConfirmation` or `UnsubscribeConfirmation`) it returns `{}`. When it returns `{}`, it logs an annoying validation error.

That said, my comparison is wrong and should check for `parsedMessage != undefined` instead of just `parsedMessage`, just in case we do get something falsy as the parseMessage.